ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/transaction/UtxoLedgerTransaction.kt

```diff
@@ -36,6 +38,16 @@ interface UtxoLedgerTransaction {
      */
     val id: SecureHash
 
+    /**
+     * @property notary The notary used for notarising this transaction.
+     */
+    val notary: Party
+
+    /**
+     * @property metadata The transaction's metadata.
+     */
+    val metadata: TransactionMetadata
```

Comment:
Should we add the metadata to the consensual one as well?

Reply:
It seemed to be less useful for me, since ledgerTx verify is only called from two places:
1. ConsensualSignedTransactionFactoryImpl which verifies metadata already before creating the signed Tx.
2. ConsensualFinalityBase where the signedTransaction is also available.

Also the ConsensualLedgerTransaction API is quite restricted at the moment. I did not want to expose anything just for code simplicity.

But tbh I do not see a big difference, so happy to add it there if you think that is better.